src/handler.rs

```diff
@@ -2,21 +2,40 @@ use crate::flows;
 use crate::flows::core::Variant;
 use crate::structures::option::Command::{Best, Fn, Preview, Query, Repo, Search, Widget};
 use crate::structures::option::{Config, RepoCommand};
-use std::error::Error;
+use anyhow::Context;
+use anyhow::Error;
 
-pub fn handle_config(mut config: Config) -> Result<(), Box<dyn Error>> {
+pub fn handle_config(mut config: Config) -> Result<(), Error> {
     match config.cmd.as_mut() {
         None => flows::core::main(Variant::Core, config, true),
+
         Some(c) => match c {
             Preview { line } => flows::preview::main(&line[..]),
-            Query { query } => flows::query::main(query.clone(), config),
-            Best { query, args } => flows::best::main(query.clone(), args.to_vec(), config),
-            Search { query } => flows::search::main(query.clone(), config),
+
+            Query { query } => {
+                let error_string = format!("Failed to filter cheatsheets for {}", &query);
```

Comment:
Wouldn't `with_context` prevent us from generating this str 100% of the times?

Reply:
`query` is part of `config`. But `config` is given away to `flows::query::main` on :17, which makes it impossible to use `query` after that. So we need to make a copy of `query` before that happens. 

`let query_clone = query.clone()` and then passing that to `with_context` would be a possibility. Not sure how smart the compiler is. I'll change it to be sure :thinking: